### PR TITLE
[Feature] Support LDAP auth in new RBAC

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

1. Support LDAP auth in new RBAC 
2. Support show user property statement in new RBAC 
3. Support show authentication statement in new RBAC 

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Review by Astralidea — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 36 / 100 (36.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/ShowUserPropertyStmt.java|0|7|00.00%|[63, 64, 65, 68, 69, 70, 71]|
|:large_blue_circle:|com/starrocks/authentication/UserAuthenticationInfo.java|0|1|00.00%|[93]|
|:large_blue_circle:|com/starrocks/authentication/LDAPAuthenticationProvider.java|2|17|11.76%|[29, 30, 31, 32, 38, 39, 40, 42, 44, 49, 50, 51, 52, 53, 54]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|26|66|39.39%|[334, 335, 336, 338, 339, 340, 342, 343, 344, 346, 347, 348, 349, 350, 351, 352, 353, 355, 356, 357, 358, 359, 468, 469, 523, 525, 527, 945, 1104, 1202, 1224, 1764, 1793, 1798, 1811, 1817, 1830, 1835, 1848, 1853]|
|:large_blue_circle:|com/starrocks/authentication/AuthenticationManager.java|4|5|80.00%|[502]|
|:large_blue_circle:|com/starrocks/mysql/privilege/AuthUpgrader.java|4|4|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java [hunk 1]
```diff
@@ -327,9 +331,33 @@ public ShowResultSet execute() throws AnalysisException, DdlException {
...
         final ShowAuthenticationStmt showAuthenticationStmt = (ShowAuthenticationStmt) stmt;
-        List<List<String>> rows = GlobalStateMgr.getCurrentState().getAuth().getAuthenticationInfo(
-                showAuthenticationStmt.getUserIdent());
-        resultSet = new ShowResultSet(showAuthenticationStmt.getMetaData(), rows);
+        if (connectContext.getGlobalStateMgr().isUsingNewPrivilege()) {
+            AuthenticationManager authenticationManager = GlobalStateMgr.getCurrentState().getAuthenticationManager();
+            List<List<String>> userAuthInfos = Lists.newArrayList();
+
+            Map<UserIdentity, UserAuthenticationInfo> authenticationInfoMap = new HashMap<>();
+            if (showAuthenticationStmt.getUserIdent() == null) {
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:339]: When null, we should check showAuthenticationStmt.isAll(), if is true, return all users info, if not, only return the auth info of current user, i think the original logic in `Auth` is wrong.

BTW, the privilege check of visitShowAuthenticationStatement() in PrivilegeCheckerV2 is incorrect, plz also modify that function.

image

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java [hunk 2]
```diff
@@ -327,9 +331,33 @@ public ShowResultSet execute() throws AnalysisException, DdlException {
...
+            for (Map.Entry<UserIdentity, UserAuthenticationInfo> entry : authenticationInfoMap.entrySet()) {
+                UserAuthenticationInfo userAuthenticationInfo = entry.getValue();
+                userAuthInfos.add(Lists.newArrayList(
+                        entry.getKey().toString(),
+                        userAuthenticationInfo.getPassword().length == 0 ? "No" : "Yes",
+                        userAuthenticationInfo.getAuthPlugin(),
+                        userAuthenticationInfo.getTextForAuthPlugin()));
+            }
+
+            resultSet = new ShowResultSet(showAuthenticationStmt.getMetaData(), userAuthInfos);
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:355]: plz add ut for `show [ALL] authentication [for user]`

File: fe/fe-core/src/main/java/com/starrocks/authentication/LDAPAuthenticationProvider.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        String userForAuthPlugin = authenticationInfo.getTextForAuthPlugin();
+        if (!Strings.isNullOrEmpty(userForAuthPlugin)) {
+            LdapSecurity.checkPassword(userForAuthPlugin, new String(password));
+        } else {
+            LdapSecurity.checkPasswordByRoot(user, new String(password));
+        }
+    }
+
+    @Override
+    public UserAuthenticationInfo upgradedFromPassword(UserIdentity userIdentity, Password password)
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/authentication/LDAPAuthenticationProvider.java:47]: Have we tested upgrade ldap authentication from old version to new version(isUsingNewPrivilege = true) manually?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 40 / 103 (38.83%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/ShowUserPropertyStmt.java|0|7|00.00%|[63, 64, 65, 68, 69, 70, 71]|
|:large_blue_circle:|com/starrocks/authentication/UserAuthenticationInfo.java|0|1|00.00%|[93]|
|:large_blue_circle:|com/starrocks/authentication/LDAPAuthenticationProvider.java|2|17|11.76%|[29, 30, 31, 32, 38, 39, 40, 42, 44, 49, 50, 51, 52, 53, 54]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|30|69|43.48%|[335, 336, 338, 339, 340, 343, 344, 345, 347, 348, 350, 352, 353, 354, 355, 356, 357, 358, 359, 361, 362, 372, 373, 486, 487, 541, 543, 545, 848, 905, 1117, 1122, 1134, 1610, 1816, 1835, 1848, 1866, 1871]|
|:large_blue_circle:|com/starrocks/authentication/AuthenticationManager.java|4|5|80.00%|[502]|
|:large_blue_circle:|com/starrocks/mysql/privilege/AuthUpgrader.java|4|4|100.00%|[]|

**Review by nshangyiming — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by HangyuanLiu** [comment]
run starrocks_admit_test

**Comment by nshangyiming** [comment]
run starrocks_admit_test

**Comment by HangyuanLiu** [comment]
run starrocks_admit_test

**Review by gengjun-git — APPROVED** [review]

**Pull request merged**